api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationOAuth.java

```diff
@@ -141,6 +168,26 @@ public void setUserNameClaim(String userNameClaim) {
         this.userNameClaim = userNameClaim;
     }
 
+    @Description("The fallback username claim to be used for user id if the claim specified by `userNameClaim` is not present. " +
```

Comment:
Why is this whole fallback structure needed? Maybe we should explain a bit more in the description when to use it?

Reply:
This is to support some fringe cases - it's hard to tell how commonly it may be used in the end. There are authorization server implementations that set `preferred_username` or similar into JWT token only if `password` authentication was used to obtain the token from token endpoint (that's using username and password). If `client_credentials` authentication  is used (that's using clientId and clientSecret), they set `client_id` or similar into JWT token. If we only have userNameClaim we would have to set it to either `preferred_username` or to `client_id`. In the first case none of the clients would have a user id in the second none of the users would have a user id. Without user id we can say that the authorization server trusts this user, but on Kafka broker we don't have an identifier what could serve for ACLs or  Keycloak authorization.

Reply:
Makes sense. Thanks for the explanation.

Reply:
Added more explanation to asciidoc documentation.